### PR TITLE
rpmsg_virtio: Move get RX buffer into receive loop and returned buffer kick to common location 

### PR DESCRIPTION
**Pull request by glneo**

### CODE REVIEW

**Review by tnmysh — COMMENTED** [review]
>  This should have
> no functional change but prepares for a later bug fix.

As commit message says this is needed for bug fix, then this change can be squashed in same commit where actual bug is fixed.

**Comment by glneo** [comment]
Okay, the bug fix I have in mind has a few more dependencies so for now I'll remove that part of the commit message. The patch should stand on its own just the same.

File: lib/rpmsg/rpmsg_virtio.c [hunk 1]
```diff
@@ -569,14 +569,16 @@ static void rpmsg_virtio_rx_callback(struct virtqueue *vq)
 	uint16_t idx;
 	int status;
 
-	metal_mutex_acquire(&rdev->lock);
-
-	/* Process the received data from remote node */
-	rp_hdr = rpmsg_virtio_get_rx_buffer(rvdev, &len, &idx);
+	while (true) {
```
Comment (arnopo) [inline lib/rpmsg/rpmsg_virtio.c:572]: nitpicking, to be aligned with the other infinite loop in the lib 
```suggestion
	while (1) {
```

**Review by arnopo — COMMENTED** [review]
Minor remark else LGTM

File: lib/rpmsg/rpmsg_virtio.c (hunk 1 above)
Comment (glneo) [reply lib/rpmsg/rpmsg_virtio.c]: Sure, changed.

**Review by arnopo — APPROVED** [review]

**Review by tnmysh — APPROVED** [review]
LGTM.

**Pull request merged**